src/Components/GEOSldas_GridComp/GEOS_LdasGridComp.F90

```diff
@@ -620,22 +620,43 @@ subroutine Initialize(gc, import, export, clock, rc)
        close(10)
        call io_grid_def_type('w', logunit, tile_grid_f, 'tile_grid_f')
 
+       ! get a grid for perturbations and EnKF:
+       !
+       !  tile grid                 !  pert grid
+       !  (defines tile space)      !  (used for perturbations and as "hash" grid in EnKF analysis)
+       ! ===========================================================================================================
+       !  lat/lon                   !  same as tile_grid (i.e., lat/lon)
+       ! -----------------------------------------------------------------------------------------------------------
+       !  EASEv[X]                  !  same as tile_grid (i.e., EASE)
+       ! -----------------------------------------------------------------------------------------------------------
+       !  cube-sphere               !  lat/lon grid of resolution similar to that of (cube-sphere) tile_grid
+       
        pert_grid_g = get_pert_grid(tile_grid_g)
-       pert_grid_f = get_pert_grid(tile_grid_f)
-
-       if(trim(grid_type) == "Cubed-Sphere" ) then
+       ! pert_grid_f = get_pert_grid(tile_grid_f) -- SEEMS REDUNDANT, OVERWRITTEN BELOW: pert_grid_f = get_minExtent_grid()
+       
+       !if(trim(grid_type) == "Cubed-Sphere" ) then
+       !
+       !   _ASSERT(index(tile_grid_g%gridtype, 'c3') /=0, "tile_grid_g does not describe a cubed-sphere grid")
+       
+       if ( .not. (pert_grid_g==tile_grid_g) ) then
```

Comment:
@weiyuan-jiang: Note the change in the if condition here.  The tasks within the if block need to get done whenever pert_grid differs from tile_grid.  I don't think we should make this conditional on the CubeSphere grid.  It just happens that pert_grid differs from tile_grid when we're in CubeSphere tile space.